### PR TITLE
refactor(docs): GCP NAT Example update

### PR DESCRIPTION
**Pull request by jameswinegar**

### CODE REVIEW

**Review by jamilbk — APPROVED** [review]
LGTM, thanks for the contribution.

File: terraform/examples/gcp/nat_gateway/network.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
+resource "google_compute_network" "firezone" {
```
Comment (AndrewDryga) [inline terraform/examples/gcp/nat_gateway/network.tf:1]: @jamilbk would it be better if we just accept a network ID and use `default` (that all projects have) if it's not set? I think if somebody deploys it to a VPC the network might already exist so creating a new one is not always desired. Or you did it this way not to enable NAT on default subnet?

**Review by AndrewDryga — APPROVED** [review]

**Comment by AndrewDryga** [comment]
Hello @jameswinegar, can you please take a look at #4183 and let me know if you think that there is something missing from it? I'm sorry for replacing your PR but I think we should use regional groups sooner than later to make sure that our examples allow zero-downtime deployments.

**Comment by AndrewDryga** [comment]
Closing the PR because #4183 is merged and contains both changes made here along with zero-downtime upgrades and better module reuse.

**Pull request closed without merging**